test/builtin/test_datentime.py

```diff
@@ -8,17 +9,28 @@
 
 
 @pytest.mark.skipif(
-    sys.platform in ("win32",) or hasattr(sys, "pyston_version_info"),
+    sys.platform in ("win32",),  # or hasattr(sys, "pyston_version_info"),
+    reason="pyston and win32 does not do well killing threads",
+)
+def test_timeremaining0():
+    str_expr = "TimeConstrained[Integrate[Sin[x]^1000000, x], 0.9]"
```

Comment:
Do we really need something complicated and something that takes a lot of time like this? 

Maybe something like `TimeConstrained[1 + 2, 0.00001]` would work and would be a lot faster. 

(I tried in current master and it doesn't work. In that I learn we don't have a `$TimeUnit`.

Reply:
I tried `TimeConstrained[Integrate[Sin[x]^10000, x], 0.5]` and that give Aborted too. This is simpler and faster. Can we get this the exponent smaller and the delay time smaller still?

Reply:
The goal of this test is to check if the thread is killed, or not. Ideally, you reach the wall time before you saturates the resources and rise a segmentation fault.

Reply:
And if something goes wrong, you leave the developer with a long-running mess.

Reply:
The idea is to merge this into the point in which we are sure that such a long-running mess does not happen. But to see that the evaluation was really stopped, I need something that complains alone. 

Another possible test would be
```a=0;TimeConstrained[Do[Pause[.1];a=a+1,{10000}], .1];a<10``` 
but the problem with it is that almost all the evaluation happens inside Mathics, where we are checking the status of the `timeout` flag. 

What I am trying to test with this expression are evaluations that happen in sympy or numpy, where we do not have a way to fix a wall time.

Reply:
> I tried `TimeConstrained[Integrate[Sin[x]^10000, x], 0.5]` and that gives Aborted too. This is simpler and faster. Can we get this the exponent smaller and the delay time smaller still?

I see that instead of 0.5, a value of 0 has to be the best.

Reply:
Another possibility that is easily done in pytest is to have long-running tests performed via an OS Environment variable. Then we can arrange a CI workflow (one in parallel) that doesn't run the entire suite but just the choice long-running tests.